ADR/user-specified-tokens-build.md

```diff
@@ -0,0 +1,28 @@
+# Enable users to configure tokens/credentials for a Build/Test Pipeline
+
+Date: 2023-03-03
+
+## Status
+
+Provisional
+
+
+## Context
+
+As a user, I would like to configure sensitive information/tokens/credentials for the build/test pipeline to consume as part of a partner task.
+
+
+## Decision
+
+1. UI will take in sensitive information as form input and upload it to SPI using the user's authentication token. See https://github.com/redhat-appstudio/service-provider-integration-operator/blob/main/docs/USER.md#uploading-access-token-to-spi-using-kubernetes-secret .
+2. In the above form, the user should be able to choose the name of the `Secret` from a dropdown. 
+The dropdown would be a pre-populated list of `Secret` names associated with known partner integrations. The user should also be allowed to provide a name barring the ones that already exist in the namespace.
```

Comment:
Building on an earlier question about whether these are just for partner tasks, would a user be able to enter a freeform secret name if they were able to leverage them in their integration tests?

Reply:
Since the `Secret` name is more of a runtime implementation detail, I reworked the ADR to say that users shouldn't really have any say in the name of the `Secret` - they should only have a say on what the associated parameter name is ( and the content of the `Secret` of course ). The UI should definitely notify the user what the name of the `Secret` is though, so that the user is able to reference it in the customized Pipeline. What do you think ?

Reply:
As long as there is a clear user direction between the secret need (i.e. partner task), setting, and using (i.e. in integration tests) then I think we should be good. 

As an example, after configuring a secret (or if we can see the secrets that are set for the workspace), then we can get a display bubble showing something to the effect of "In order to leverage this secret in a task, do X or Y."

Looking at the proposed changes, I think this relationship (as well as the irrelevance of the secret name) makes sense.